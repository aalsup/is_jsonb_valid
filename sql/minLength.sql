-- minLength validation
-- longer is valid
SELECT is_jsonb_valid('{"minLength":2}', '"foo"');
-- exact length is valid
SELECT is_jsonb_valid('{"minLength":2}', '"fo"');
-- too short is invalid
SELECT is_jsonb_valid('{"minLength":2}', '"f"');
-- ignores non-strings
SELECT is_jsonb_valid('{"minLength":2}', '1');
-- one supplementary Unicode code point is not long enough
SELECT is_jsonb_valid('{"minLength":2}', '"ðŸ’©"');