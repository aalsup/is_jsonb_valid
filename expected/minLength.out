-- minLength validation
-- longer is valid
SELECT is_jsonb_valid('{"minLength":2}', '"foo"');
 is_jsonb_valid 
----------------
 t
(1 row)

-- exact length is valid
SELECT is_jsonb_valid('{"minLength":2}', '"fo"');
 is_jsonb_valid 
----------------
 t
(1 row)

-- too short is invalid
SELECT is_jsonb_valid('{"minLength":2}', '"f"');
 is_jsonb_valid 
----------------
 f
(1 row)

-- ignores non-strings
SELECT is_jsonb_valid('{"minLength":2}', '1');
 is_jsonb_valid 
----------------
 t
(1 row)

-- one supplementary Unicode code point is not long enough
SELECT is_jsonb_valid('{"minLength":2}', '"ðŸ’©"');
 is_jsonb_valid 
----------------
 f
(1 row)

